### DISCUSSION TITLE
when CWA is not an NWS office code

### DISCUSSION THREAD

**Discussion by JJKraw** [weather-gov/api#550] [Q&A] [answered]
For most land forecast zones, the CWA for that zone matches the office code that is responsible for it. E.g., for

https://api.weather.gov/zones/forecast/MAZ014

```
        "cwa": [
            "BOX"
        ],
        "forecastOffices": [
            "https://api.weather.gov/offices/BOX"
        ],
```

In turn, the office contains the URLs of the counties and zones for which it is responsible. I.e., if you go to the forecastOffices link, a link back to MAZ014 exists under responsibleForecastZones. This same relationship seems to hold for coastal marine zones as well.

However, for offshore marine zones, as well as the zones that were recently decoupled from the Guam office, this is not true. The CWA code does not respond to an office, yet it is used in the forecastOffices link.

https://api.weather.gov/zones/forecast/AMZ079

```
        "cwa": [
            "NH2"
        ],
        "forecastOffices": [
            "https://api.weather.gov/offices/NH2"
        ],
```

If one follows the link to NH2, the return is a 404:

```
{
    "correlationId": "83a31a",
    "title": "Office Does Not Exist",
    "type": "https://api.weather.gov/problems/InvalidOffice",
    "status": 404,
    "detail": "Office NH2 does not exist",
    "instance": "https://api.weather.gov/requests/83a31a"
}
```

For the former Guam CWA zones:

https://api.weather.gov/zones/forecast/MHZ002

```
        "cwa": [
            "PQE"
        ],
        "forecastOffices": [
            "https://api.weather.gov/offices/PQE"
        ],
```

https://api.weather.gov/offices/PQE

```
{
    "correlationId": "1eefcd89",
    "parameterErrors": [
        {
            "parameter": "path.officeId",
            "message": "Does not have a value in the enumeration [\"AKQ\",\"ALY\",\"BGM\",\"BOX\",\"BTV\",\"BUF\",\"CAE\",\"CAR\",\"CHS\",\"CLE\",\"CTP\",\"GSP\",\"GYX\",\"ILM\",\"ILN\",\"LWX\",\"MHX\",\"OKX\",\"PBZ\",\"PHI\",\"RAH\",\"RLX\",\"RNK\",\"ABQ\",\"AMA\",\"BMX\",\"BRO\",\"CRP\",\"EPZ\",\"EWX\",\"FFC\",\"FWD\",\"HGX\",\"HUN\",\"JAN\",\"JAX\",\"KEY\",\"LCH\",\"LIX\",\"LUB\",\"LZK\",\"MAF\",\"MEG\",\"MFL\",\"MLB\",\"MOB\",\"MRX\",\"OHX\",\"OUN\",\"SHV\",\"SJT\",\"SJU\",\"TAE\",\"TBW\",\"TSA\",\"ABR\",\"APX\",\"ARX\",\"BIS\",\"BOU\",\"CYS\",\"DDC\",\"DLH\",\"DMX\",\"DTX\",\"DVN\",\"EAX\",\"FGF\",\"FSD\",\"GID\",\"GJT\",\"GLD\",\"GRB\",\"GRR\",\"ICT\",\"ILX\",\"IND\",\"IWX\",\"JKL\",\"LBF\",\"LMK\",\"LOT\",\"LSX\",\"MKX\",\"MPX\",\"MQT\",\"OAX\",\"PAH\",\"PUB\",\"RIW\",\"SGF\",\"TOP\",\"UNR\",\"BOI\",\"BYZ\",\"EKA\",\"FGZ\",\"GGW\",\"HNX\",\"LKN\",\"LOX\",\"MFR\",\"MSO\",\"MTR\",\"OTX\",\"PDT\",\"PIH\",\"PQR\",\"PSR\",\"REV\",\"SEW\",\"SGX\",\"SLC\",\"STO\",\"TFX\",\"TWC\",\"VEF\",\"AER\",\"AFC\",\"AFG\",\"AJK\",\"ALU\",\"GUM\",\"HPA\",\"HFO\",\"PPG\",\"STU\",\"NH1\",\"NH2\",\"ONA\",\"ONP\"]"
        }
    ],
    "title": "Not Found",
    "type": "https://api.weather.gov/problems/NotFound",
    "status": 404,
    "detail": "Not Found",
    "instance": "https://api.weather.gov/requests/1eefcd89"
}
```

I'll also note that an office query using any of the values listed after PPG in the message also results in a 404.

So anyway... I'm just wondering how this is all _supposed_ to work. I assume bad URLs aren't a goal :) and also that if an error message contains what is supposed to be a valid enumeration, it shouldn't lead to a 404.

Thanks a lot!!

**Comment by StephenClouse** [comment, marked as answer]
Yes, it's a mess because the GIS group keeps overloading what the "CWA" field actually means in the source data.

All I can say is we have a bunch of internal tickets on these issues but no timetable for fixes.

**Comment by JJKraw** [comment]
Thanks, Stephen. Glad to know it's not just my usual being dense. ;)

Best of luck on this!

**Comment by JJKraw** [comment]
A follow-up question if I may, rather than creating another new thread...

There is currently a small craft advisory for American Samoa waters (area PS) which doesn't appear in the API.

https://alerts.weather.gov/cap/wwacapget.php?x=MZPI1263EF6EE578.SmallCraftAdvisory.1263EF70F660MZPI.STUCWFPPG.bc5112f6111ce10324d78295d946f13f

Is this an operational issue that should be reported or is there some bigger ingestion issue with American Samoa?

**Reply by StephenClouse** [reply]
It's hard to say when comparing with the legacy alerts site. Some of those alerts are from upstream, and some are actually generated from text products by the software itself.

API strictly uses the CAP 1.2 feed from NCEP IDP CAP Handler. One sticking point with that (which has prevented the full cut-over to the newer CAP-1.2-based alerts-v2 site) is that certain products simply aren't there. I know Alaska marine is one. It wouldn't surprise me if Samoa marine was missing as well.

Raising it as an ops ticket wouldn't hurt.

**Reply by scadergit** [reply]
I was told yesterday that my next work will be getting https://alerts-v2.weather.gov ready to replace the current alerts. That version is solely based on the API, which gives me hope that NWS is reconciling that issue.

**Reply by StephenClouse** [reply]
Well, that's good to know. I haven't had time to keep up on it. Have my own set of twelve projects to juggle 😛

**Reply by JJKraw** [reply]
> API strictly uses the CAP 1.2 feed from NCEP IDP CAP Handler. One sticking point with that (which has prevented the full cut-over to the newer CAP-1.2-based alerts-v2 site) is that certain products simply aren't there. I know Alaska marine is one. It wouldn't surprise me if Samoa marine was missing as well.
> 

Alaska Marine used to not work but I've been pulling those with no problem for a while.

  https://api.weather.gov/alerts/active/area/PK

I'll pop an email to ops about Samoa.

**Comment by JJKraw** [comment]
Here is what I heard back regarding America Samoa:

> OBT reached out to Tier3 support for CAP Handler to check on this and CAP alerts are not created for American Samoa, thus they are not available from API or in the wwa shapefile produced for radar/idpgis.

I assume someone gets to figure out how this eventually gets into the API.
